dashboard/app/models/stage.rb

```diff
@@ -14,6 +14,7 @@
 # (Intended to replace most of the functionality in Game, due to the need for multiple app types within a single Game/Stage)
 class Stage < ActiveRecord::Base
   has_many :script_levels, -> { order('position ASC') }, inverse_of: :stage
+  has_one :plc_learning_module, class_name: 'Plc::LearningModule', dependent: :destroy
```

Comment:
Without this, changing the name of a stage will result in two `Plc::LearningModule` objects (the old one and the new).  Also this is how level naming works today — if you change the name of a level, all progress for that level will be lost:

> Note: renaming a level is essentially making a copy and deleting the old one. This means that progress in that level is lost, so don't do this for published scripts. If you do rename a level you need to update all scripts referencing it, this does not happen automatically.

Adding a GUID to `Stage` could solve this, but for now the workaround will be to rename stages using `scripts.en.yml` which is already the best practice.